tests/moduletool/test_update.py

```diff
@@ -183,6 +183,45 @@ def assert_version_installed(module_name: str, version: str) -> None:
     assert ModuleV1(project=None, path=mod11_dir).version == Version("4.1.2")
 
 
+@pytest.mark.parametrize("install_mode", [InstallMode.release, InstallMode.prerelease])
+@pytest.mark.slowtest
+def test_module_update_with_v1_module(
+    tmpdir: py.path.local,
+    modules_dir: str,
+    snippetcompiler_clean,
+    modules_repo: str,
+    install_mode: InstallMode,
+) -> None:
+    """
+    Assert that the ast cache is invalidated after a module is installed. This so that dependencies of
+    installed modules are updated as well.
+
```

Comment:
I'm not 100% sure this test is actually checking that the cache is invalidated i.e. I would expect this test to fail if we comment out the cache invalidation in `src/inmanta/moduletool.py`

Reply:
Indeed, it should fail. good catch. i'll continue looking

Reply:
@arnaudsjs I don't think the cache invalidation is needed for v1 modules. When calling `my_project.collect_requirements()` the new requirements after each module update/install are updated and it manages to install everything

Reply:
The cache invalidation is required for v1 modules, but the test scenario doesn't trigger the issue. The difference between this test case and the one for v2 modules is that this one adds an import statement for mod11 to mod13 both in the initially installed version of the module and the version installed after the update. The test case for v2 doesn't have the import statement in the initially installed version. The moduletool uses the import statements to determine the full set of modules that are required by the project. The idea behind the test is that only the new module version has the import statement and that the cache causes the moduletool to still see the version without the import statement, causing it not to update the version of the module dependency.